pkg/tagger/collectors/ecs_common_test.go

```diff
@@ -8,24 +8,80 @@ package collectors
 import (
 	"testing"
 
+	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/tagger/utils"
+
 	"github.com/stretchr/testify/assert"
 )
 
-func TestAddTaskResourceTags(t *testing.T) {
-	tags := utils.NewTagList()
-	taskTags := map[string]string{
-		"environment":         "sandbox",
-		"project":             "ecs-test",
-		"aws:ecs:clusterName": "test-cluster",
-		"aws:ecs:serviceName": "nginx-awsvpc",
+func TestAddResourceTags(t *testing.T) {
+	tests := []struct {
+		name      string
+		taskTags  map[string]string
+		loadFunc  func() *utils.TagList
+		resetFunc func()
+	}{
+		{
+			name: "nominal case",
+			taskTags: map[string]string{
+				"environment":         "sandbox",
+				"project":             "ecs-test",
+				"aws:ecs:clusterName": "test-cluster",
+				"aws:ecs:serviceName": "nginx-awsvpc",
+			},
+			loadFunc: func() *utils.TagList {
+				expectedTags := utils.NewTagList()
+				expectedTags.AddLow("environment", "sandbox")
+				expectedTags.AddLow("project", "ecs-test")
+				return expectedTags
+			},
+			resetFunc: func() {},
+		},
+		{
+			name: "replace colon enabled",
```

Comment:
In your test you'll want to cover the case where a tag _value_ has a colon: in this case, even with the option enabled, the colon in the tag value should not be replaced

Reply:
makes sense, thanks!